.github/scripts/apidoc.sh

```diff
@@ -0,0 +1,158 @@
+#!/usr/bin/env bash
+#
+# Process (API) docs after a successful build on Travis (via ../.travis.yml).
+#
+# Updated/changed documentation for "master" is pushed to gh-pages.
+# In case of pull requests or other branches, it will get added to a separate branch.
+# In case of a pull request, a compare view comment will be posted.
+#
+# NOTE: stdout/stderr might/should be discarded to not leak sensitive information.
+
+echo "Post-processing (API) documentation."
+echo "PR Number: $PR_NUMBER"
+echo "GITHUB_HEAD_REF: $GITHUB_HEAD_REF"
+
+if [ -z "$APIDOC_TOKEN" ]; then
+    echo "No APIDOC_TOKEN available. Skipping."
+    exit
+fi
+
+# NOTE: DO NOT USE "set -x", or anything else that would reveal APIDOC_TOKEN!
+# GitHub Actions does filter secrets, but extra caution won't hurt
+set -e
+set +x
+
+# Display exit code in case of failure (probably due to 'set -x').
+trap '[ "$?" = 0 ] || echo "EXIT CODE: $?"' EXIT
```

Comment:
This file was moved from `build-utils/travis-apidoc.sh` (and looking at the comments below, I missed some references). Maybe GitHub wasn't able to display that accurately.

All code that isn't specific to Travis/GitHub Actions was left as-is, since this can't be fully tested in a fork. Some parts require write access to https://github.com/awesomeWM/apidoc, so I was only able to run it with some NOOPs.
I'll implement the formatting suggestions, but the suggestions related to functionality should be done in a PR that has the required access to properly test such changes.

Reply:
This was added in 30eafb113.
@ShayAgros 
At least it should have said `probably due to 'set -e'`, not `set -x`.
I think it achieves what it wanted: to display the exit code if it is not zero, no?
Apart from that I think it could be fixed up (in a separate patch/PR).

Reply:
First of all sorry for not being more involved in this PR. I hope this would change soon.

@blueyed I'm not sure what the code actually tries to achieve. I'd guess that it was added to give more information about why the script failed by tapping on the exit signal. Using `set -e` would make more sense here imo, not sure why it didn't.

The problem with this line is that by the time `echo "EXIT CODE: $?` is executed the value of `?` is the one returned from the expression `[ "$?" = 0 ]` and not by the failed command. The snippet below should have returned 127 which is the error code of unknown command, but instead it returns 1:
```bash
set -e

trap '[ "$?" = 0 ] || echo "EXIT CODE: $?"' EXIT

dsajkld
```

So printing the exit code here doesn't add information. The snippet I suggested traps the error code correctly:
```bash
set -e

trap 'echo EXIT CODE: $?' ERR

dsajkld
```
returns:
```
./test.sh: line 8: dsajkld: command not found
EXIT CODE: 127
```